File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-4.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    findMin(nums) {
        let start = 0;
        let end = nums.length-1;
        let mid =0;
        while(start <= end){
            mid = Math.trunc((start + end)/2);
            console.log("mid : ",nums[mid]);
            if(nums[mid]>nums[nums.length-1]){
                start = mid+1;
            }else{
                if(nums[mid]<nums[mid-1]){
                    return nums[mid];
                }
                end = mid-1;
            }
        }
        return nums[mid];
    }
}

File: Data Structures & Algorithms/jump-game-ii/submission-4.java
class Solution {
    public int jump(int[] nums) {
        int l=0,r=0,res=0;
        int farthest = 0;
        int len = nums.length;
        while(r<nums.length-1){
            farthest = 0;
            for(int i=l; i<=r; i++){
                farthest = Math.max(farthest,i+nums[i]);
            }
            res++;
            l=r+1;
            r=farthest;
        }
        return res;
    }   
}
